src/label-helpers.ts

```diff
@@ -22,37 +25,46 @@ function getTextContent(node) {
     .join('')
 }
 
-function getLabelContent(node) {
+function getLabelContent(node: Node | Element | HTMLInputElement) {
   let textContent
-  if (node.tagName.toLowerCase() === 'label') {
+  if ('tagName' in node && node.tagName.toLowerCase() === 'label') {
     textContent = getTextContent(node)
+  } else if ('value' in node) {
+    textContent = node.value
   } else {
-    textContent = node.value || node.textContent
+    textContent = node.textContent
   }
   return textContent
 }
 
 // Based on https://github.com/eps1lon/dom-accessibility-api/pull/352
-function getRealLabels(element) {
-  if (element.labels !== undefined) return element.labels ?? []
+function getRealLabels(element: Element | HTMLInputElement) {
+  // for old browsers
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  if ('labels' in element && element.labels !== undefined)
+    return element.labels ?? []
```

Comment:
I would prefer we don't change any runtime behavior but rather override the typechecker with type-casting or `@ts-expect-error`. Right now I can't know from the review whether this changes runtime behavior, or was unsound, or just TS being overly strict.

Reply:
The problem here is the labels can be null and not undefined. Btw I think that on old browser like ie labels is undefined and not null

Reply:
Looks to me covered all these cases and TypeScript is not able to narrow. Let's keep the runtime and annotate the compile time properly.

Reply:
The problem is that the linter is giving the problem, so `@ts-expect-error` will not work

Reply:
Ignore the linter.